### PR TITLE
Feature/create resource

### PR DESCRIPTION
**Pull request by ilija-savic-devtech**
# What's the purpose of this MR?
Creating a resource in database
# Impact Analysis
(N/A)
# Where should the tester start?
- Run start_server.bat
- Run Postman
- When you are in Postman, go to body tab and for key values enter: name, surname, indexno, address. For values enter what you want.
- After that, go to field where you can type address and type: localhost/create
- Chose POST method and click SEND button
# How should this be manually tested?
(N/A)
# Any background context you want to provide?
(N/A)
# What are the relevant tickets?
(N/A)
# Screenshots (if appropriate)
(N/A)

### CODE REVIEW

File: public/index.php [hunk 1]
```diff
@@ -16,6 +16,14 @@
 	));
 });
 
+$klein->respond('/create', function() use ($twig, $crud){
```
Comment (goran-erhartic-devtech) [inline public/index.php]: please checkout Klein documentation on how to properly define routes with methods
https://github.com/klein/klein.php

do the same for all routes, not just this one

File: public/index.php [hunk 2]
```diff
@@ -16,6 +16,14 @@
...
+	$crud->create($_POST['name'], $_POST['surname'], $_POST['indexno'], $_POST['address']);
```
Comment (goran-erhartic-devtech) [inline public/index.php]: why pass four parameters, when we can pass only one? ;)

File: public/index.php [hunk 3]
```diff
@@ -16,6 +16,14 @@
...
+	$getAll = $crud->getAll();
```
Comment (goran-erhartic-devtech) [inline public/index.php]: we do not need this in our 'create' route nor do we need to show the end result, we can call this method after we have created our student

File: public/index.php [hunk 4]
```diff
@@ -16,6 +16,14 @@
...
+	echo $twig->render('home.twig', array(
```
Comment (goran-erhartic-devtech) [inline public/index.php]: instead all of this, you can simply let the user know that 'student' has been created - if operation was successful, or if not, what went wrong

**Comment by goran-erhartic-devtech** [comment]
Btw, you can use 'MongoConnectionException' class to handle mongodb connection exception in your 'DbConnection' class

**Comment by ilija-savic-devtech** [comment]
I tried to use 'MongoConnectionException' but it won't to catch any exception when server is down. I used this in my connectMongoDb function in DbConnection class:
try {
			$this->conn = new Manager(MONGODB_URI);

			return $this->conn;
		} catch(\MongoConnectionException $e){
			echo "Connection failed: " . $e->getMessage();
		}

File: public/index.php (hunk 1 above)
Comment (ilija-savic-devtech) [reply public/index.php]: Something like this for all routes?
$klein->respond('GET','/', function() use ($twig, $crud){
	$getAll = $crud->getAll();
	echo $twig->render('home.twig', array(
		'getAll' => $getAll
	));
});

Comment (goran-erhartic-devtech) [reply public/index.php]: that is correct, for this case - method name needs to correspond to the action you want to achieve 

here is a nice summation of most commonly used methods:
http://www.restapitutorial.com/lessons/httpmethods.html

**Comment by goran-erhartic-devtech** [comment]
Mongo server needs to be up and running in order for it to work, but the exception will be caught for example if you would try to use some other hostname/port

**Comment by ilija-savic-devtech** [comment]
I did it!!! I made autoincrementer for MongoDb. Check it out, and little advice: this piece of code is my child, so choose your words carefully Persian 💯 + 💯  + 💯  😆

**Comment by ilija-savic-devtech** [comment]
Do you mean in index.php?
This piece of code:
$klein->respond('POST','/create', function() use ($crud){
	try {
		$crud->create();
		echo "Resource successfully created";
	} catch(Exception $e){
		echo "Error: " . $e->getMessage();
	}
});

**Comment by ilija-savic-devtech** [comment]
Yes it is :)

File: public/index.php [hunk 5]
```diff
@@ -2,20 +2,28 @@
...
 
-$klein->respond('/[i:id]', function($request) use($twig, $crud){
+$klein->respond('GET','/[i:id]', function($request) use($twig, $crud){
 	$getOne = $crud->getOne($request->id);
 	echo $twig->render('getOne.twig', array(
 		'getOne' => $getOne
 	));
 });
 
+$klein->respond('POST','/create', function() use ($crud){
```
Comment (goran-erhartic-devtech) [inline public/index.php]: to remain in RESTful mode, rename your route to '/', and by defining the verb POST for this route, your app will know how to handle it

**Comment by ilija-savic-devtech** [comment]
I think i done everything you asked, check it out

**Comment by goran-erhartic-devtech** [comment]
looks good 👍
proceed with delete or update

**Pull request merged**